src/util/timemanager.cpp

```diff
@@ -403,10 +421,14 @@ void TimeManager::clearKeyframes() {
 }
 
 void TimeManager::setTimeNextFrame(Time t) {
+    OpenSpaceEngine::Mode m = global::openSpaceEngine->currentMode();
+    if (global::openSpaceEngine->currentMode() == OpenSpaceEngine::Mode::CameraPath) {
+        LERROR("Cannot change simulation time during camera path");
+        return;
+    }
     _shouldSetTime = true;
     _timeNextFrame = std::move(t);
     clearKeyframes();
-    global::scriptScheduler->setCurrentTime(t.j2000Seconds());
```

Comment:
I see that you moved this to `progressTime` function. Seems like it should be fine, but I would give it a second look to make sure it wouldn't introduce any timing issues.

Reply:
I asked Malin (who added this line previously) to verify that no problems occurred by moving this line and according to her tests it did not introduce any problems